src/authenticate.js

```diff
@@ -19,24 +19,17 @@ module.exports = function(config) {
   var getServerOptions = function(path, options) {
     if (path.indexOf('http') === 0) {
       var urlObject = url.parse(path);
-      // Check if this is the format of http://project.server.com or http://project.localhost
-      var hostParts = urlObject.hostname.split('.');
-      var pathParts = urlObject.pathname.split('/');
-      // Always starts with an empty element. Throw it away.
-      if (pathParts.length > 0) {
-        pathParts.shift();
+      var pathnameParts = urlObject.pathname.split('/').filter(Boolean);
+      // If the url contains 'project' (ex. http://localhost:3000/project/63ac4f3768baf92d9bb0106f)
+      // then we extract project id from it, else - get the project name
+      if (urlObject.pathname.includes('project')) {
```

Comment:
Can we be more specific here?  Maybe do something like the following?

```
if (urlObject.pathname.match(/http[s]?:\/\/[^\/]+\/project\//)) {
```

Reply:
@travist good point, refactored